usr/share/rear/finalize/GNU/Linux/280_migrate_uuid_tags.sh

```diff
@@ -1,12 +1,86 @@
+# when restored config files have different UUIDs than those in disklayout.conf because
+# it adapts disklayout.conf UUIDs in restored config files to the actually recreated UUIDs
+# which cannot work when restored config files do not contain the disklayout.conf UUIDs.
+# The main reason for this check is that normally UUIDs get recreated as stored in disklayout.conf
+# because nowadays tools (e.g. mkfs) can set UUIDs so normally the UUID mapping code has nothing to do.
+# But when restored config files have different UUIDs than those in disklayout.conf
+# the restored config files have UUIDs that are different than the recreated UUIDs
+# so the restored config files cannot work because their UUIDs do not exist in the recreated system.
+# It is not possible to automatically adjust wrong UUIDs in restored config files with reasonable effort
+# because we know only the UUIDs in disklayout.conf and the UUIDs in the recreated system
+# but we do not know with what UUID a wrong UUID in a restored config file should be replaced
+# because we do not have the config files from the time when disklayout.conf was created.
+# The reason why restored config files can have different UUIDs than those in disklayout.conf
+# is that the backup does not match the ReaR recovery system.
+# E.g. when the backup was made at a different time than when the ReaR recovery system was made
+# (i.e. "rear mkrescue" was run at a different time than when the backup was made).
+# Using "rear mkbackup" avoids such inconsistencies but most external backup methods
+# do not support "rear mkbackup" so the user is responsible to ensure his backup and
+# his ReaR recovery system are consistent, cf https://github.com/rear/rear/issues/2787
+
+# Go to the restored files directory.
+# Careful in case of 'return' after 'pushd' (must call the matching 'popd' before 'return'):
+pushd $TARGET_FS_ROOT >/dev/null
+# Get the restored config files in CHECK_CONFIG_FILES:
+local config_files=()
+local obj
+for obj in "${CHECK_CONFIG_FILES[@]}" ; do
+    # Strip leading slash because we need relative file and directory names inside TARGET_FS_ROOT:
+    obj=${obj#/}
+    if test -d "$obj" ; then
+        config_files+=( $( find "$obj" -type f ) )
+    elif test -e "$obj" ; then
+        config_files+=( "$obj" )
+    fi
+done
+# Check if each UUID that was in at least one of the config files
+# in CHECK_CONFIG_FILES at the time when disklayout.conf was created
+# is now in at least one of the restored config files in CHECK_CONFIG_FILES:
+local uuid
+for uuid in $DISKLAYOUT_UUIDS_IN_CONFIG_FILES ; do
+    # Do not error out because at this late state of "rear recover" (after the backup was restored)
+    # but show the error to the user so after "rear recover" finished he can manually fix things as needed:
+    grep -q "$uuid" "${config_files[@]}" || LogPrintError "Error: UUID $uuid not found in a restored config file (must be manually corrected before reboot)"
```

Comment:
What if the UUID was originally (at `mkrescue` time) present in files A and B, but after restoration it is only in A and missing from B? This is likely also an error (B can be inconsistent), but user will not be warned about it.

Reply:
oh yes, and an UUID can be present multiple times in a file! Good catch.
I suppose that by "a trace of each file that will be patched" you mean its checksum or hash? If so, we have those hashes already, we just don't check them after restoration of files (only in `checklayout`).

Reply:
> also, a file might include multiple UUID to patch.

This seems to be covered in the proposed change - it checks each UUID separately.